### ISSUE TITLE
.configure script doesn't see tkinter on Arch

### ISSUE THREAD

**Issue by KhelbenArunsun** [McArcady/lnp-forge#9]
when running

`./configure --prefix=$PWD && make install`

I get this error

`checking for python... /usr/bin/python
checking python module: Tkinter... no
configure: error: failed to find required module Tkinter`

when I have the [tk](https://www.archlinux.org/packages/extra/x86_64/tk/) package installed.

should I try pip?

log for reference:
[config.log](https://github.com/McArcady/lnp-forge/files/4371882/config.log)

**Comment by McArcady** [comment]
Yes, have you tried:  
`$ pip install tkinter`  
?

**Comment by KhelbenArunsun** [comment]
Yup, first thing I tried after I verified that tk was installed. I can't figure out if the script is looking in the wrong place or what. My Python version is 3.8.

Python -m tkinter works fine, reporting version 8.6. So I have no idea what could be wrong. This is vexing.

**Comment by McArcady** [comment]
Ok could you try this ?  
in `autoconf.ac`, replace line 305:
`AC_PYTHON_MODULE([Tkinter], fatal)  `
with:
`AC_PYTHON_MODULE([tk], fatal)  `
then:
`$ autoconf && ./configure --prefix=$PWD && make install`

**Comment by KhelbenArunsun** [comment]
Yup, that worked (save that the file was named configure.ac). 

Would you mind if I edited the wiki to include the Arch dependencies? I managed to figure out which packages are equivalent to the Fedora/CentOs/Ubuntu/Debian ones.

**Comment by McArcady** [comment]
Great!
Yes please add an 'Archlinux' section on the wiki, or send me the list of dependencies

**Comment by KhelbenArunsun** [comment]
Sure, I'll close the issue, and here's the revised markdown:

`sudo pacman -Syu base-devel cmake dos2unix git help2man jdk8-openjdk lib32-zlib libxml libxslt mercurial mesa perl perl-xml-libxml perl-xml-libxslt python qt5-base qt5-tools sdl_image sdl_ttf tcl tk`

__Note:__ After installing dependencies, be sure to set your default JDK to Java 8 before you run .configure!

`sudo archlinux-java set java-8-openjdk`

**Comment by anlach** [comment]
Hey, @McArcady! I had to switch back to the python -c "import Tkinter" to get this building today.  I don't even think there is a "tk" Python module.  I'm guessing that changing the autoconf.ac effectively ignored the Tk check for @KhelbenArunsun, although I don't have a clue what his or her original error was.

**Comment by McArcady** [comment]
Thanks for the test, there seem to be 2 problems involved:
- installation of module `tkinter` with python3:  
For some reason, pip3 cannot find module tkinter, but it can still be installed with the system package manager (for instance, with Debian: `apt install python3-tk`)
- PyLNP support for python3:  
Running PyLNP (the graphical launcher of the pack) with Python3 set as default will succeed, as long as package `tkinter` has been installed, but starting the game will fail because of a python3 incompatibility in the terminal-handling code required for DFHack.

In conclusion:
- the pack produced by lnp-forge may not run if Python3 is activated by default.
- lnp-forge *should* check for `tkinter` and not only for `tk` (your fix is correct)

I fixed the python3 bug in PyLNP, but is the project still maintained ? 
Still we can activate the patch in the lnp-forge buildchain so that the next LinuxDwarfPack includes it.
